### PR TITLE
Clean up some misses in the Overlay release

### LINKED ISSUE #3564: [Bug]: Incorrect overlay positioning when overlay parent element has a CSS "transform" property applied
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-overlay, overlay-trigger

### Expected behavior

Overlay should be positioned relative to to the trigger element as rendered on screen.

### Actual behavior

When an `<sp-overlay>` or `<overlay-trigger>` element is placed within a parent element that has a CSS `transform` applied to it, the positioning of the overlay is incorrect. The placement options on the overlay can even cause the overlay to appear offscreen, making it seem like the overlay failed to open.

### Screenshots

_No response_

### What browsers are you seeing the problem in?

Firefox, Chrome, Microsoft Edge

### How can we reproduce this issue?

Create an absolutely positioned div with a CSS `transform` attribute (I used `transform: translateX(-50%)`), place an overlay trigger inside (like a button), and then place the `<sp-overlay>` element within the transformed div.

HTML
```
<div class="transformed">
  <sp-button id="trigger">Click Me</sp-button>
  <sp-overlay trigger="trigger@click" placement="top">...</sp-overlay>
</div>
```

CSS
```
.transformed {
  transform: translateX(-50%);
  position: absolute:
  left: 200px;
  top: 200px;
  width: 100px;
  height: 50px;
}
```

### Sample code that illustrates the problem

Minimal reproduction demo: https://studio.webcomponents.dev/edit/cNQsOU9R8zBOg4oZyGzB/stories/index.stories.js?p=stories

### Logs taken while reproducing problem

_No response_

### LINKED ISSUE #3587: [Bug]: Type error in MenuItem.d.ts
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

MenuItem

### Expected behavior

`import type { Overlay } from '@spectrum-web-components/overlay';` in MenuItem.ts

### Actual behavior

```
node_modules/@spectrum-web-components/menu/src/MenuItem.d.ts:6:30 - error TS2307: Cannot find module 'overlay/src/Overlay.js' or its corresponding type declarations.

6 import type { Overlay } from 'overlay/src/Overlay.js';
                               ~~~~~~~~~~~~~~~~~~~~~~~~

Found 1 error in node_modules/@spectrum-web-components/menu/src/MenuItem.d.ts:6
```
https://github.com/adobe/spectrum-web-components/blob/7fc0c2e143f20ea0aa2dca3470acf7f70e068d11/packages/menu/src/MenuItem.ts#L38

@Westbrook I think this was originally introduced in https://github.com/adobe/spectrum-web-components/commit/5dfb71e5ed26cf8af83ca335a7658938f3f135a6#diff-d6c1bef5a63290802db272d6bf262beb51675b7182fe88d7d93ceda7e32650a9
See MenuItem.ts

### Screenshots

_No response_

### What browsers are you seeing the problem in?

_No response_

### How can we reproduce this issue?

Run tsc on a project that depends on MenuItem
`./node_modules/.bin/tsc --noEmit`

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- expand overlay positioning logic
- correct type consumption

## Related issue(s)
- fixes #3564
- fixes #3587

## How has this been tested?
-   [ ] _Test case 1_
    1. Go [here](https://overlay-expanded--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-element--transformed)
    2. Confirm position of content over transformed container
-   [ ] _Test case 2_
    1. Go [here](https://overlay-expanded--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-element--contained)
    2. Confirm position of content over `contain`ed container

## Types of changes
-   [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/overlay/src/topLayerOverTransforms.ts [hunk 1]
```diff
@@ -79,9 +79,11 @@ export const topLayerOverTransforms = (): Middleware => ({
...
                 const css = getComputedStyle(containingBlock);
                 overTransforms =
-                    css.transform !== 'none' || css.filter
-                        ? css.filter !== 'none'
-                        : false;
+                    css.transform !== 'none' ||
+                    (css.filter ? css.filter !== 'none' : false) ||
+                    ['paint', 'layout', 'strict', 'content'].some((value) =>
+                        (css.contain || '').includes(value)
+                    );
```
Comment (najikahalsema) [inline packages/overlay/src/topLayerOverTransforms.ts:86]: This feels slightly esoteric to me and I'm worried we may forget what this is for 😅 It seems efficient, but maybe could benefit from a clarifying comment? Curious what others think.

**Review by najikahalsema — APPROVED** [review]
LGTM! One minor note re: clarity. I'm curious what other people think. Maybe it's fine and either way it's not blocking.

File: packages/overlay/stories/overlay-element.stories.ts [hunk 2]
```diff
@@ -152,6 +152,45 @@ longpress.args = {
     type: 'auto',
 };
 
+export const transformed = (args: Properties): TemplateResult => html`
+    <style>
+        .transformed {
```
Comment (Rajdeepc) [inline packages/overlay/stories/overlay-element.stories.ts:157]: Can we use logical CSS here too?
```
.transformed {
    transform: translateX(-50%);
    position: absolute;
    inset-inline-start: 200px; /* Equivalent to left: 200px; */
    inset-inline-end: auto; /* Resets inline-end property */
    inset-block-start: 200px; /* Equivalent to top: 200px; */
    inset-block-end: auto; /* Resets block-end property */
    width: 100px;
    height: 50px;
}
```

File: packages/overlay/stories/overlay-element.stories.ts [hunk 3]
```diff
@@ -152,6 +152,45 @@ longpress.args = {
...
+            transform: translateX(-50%);
+            position: absolute;
```
Comment (Rajdeepc) [inline packages/overlay/stories/overlay-element.stories.ts:159]: May be here too?

**Review by Rajdeepc — APPROVED** [review]
LGTM!! css change is not a blocker but a good to have

File: packages/overlay/stories/overlay-element.stories.ts (hunk 2 above)
Comment (Westbrook) [reply packages/overlay/stories/overlay-element.stories.ts:157]: Not specifically germane to the issue this confirms, but it's good practice to get into. Thanks for the reminder!

**Pull request merged**